cms/navigation/tests/test_navigation_tags.py

```diff
@@ -0,0 +1,85 @@
+from unittest.mock import Mock
+
+from django.test import TestCase
+
+from cms.navigation.templatetags.navigation_tags import main_menu_columns, main_menu_highlights
+from cms.navigation.tests.factories import (
+    ColumnBlockFactory,
+    HighlightsBlockFactory,
+    MainMenuFactory,
+    SectionBlockFactory,
+    TopicLinkBlockFactory,
+)
+
+
+class MainMenuTemplateTagTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.mock_request = Mock()
+        cls.main_menu = MainMenuFactory()
+
+        highlights = [{"type": "highlight", "value": HighlightsBlockFactory()}] * 3
+
+        # Create topic links once if they are identical across sections
+        topic_links = [TopicLinkBlockFactory()] * 5
+
+        # Create sections using the pre-defined topic links
+        sections = [SectionBlockFactory(links=topic_links)] * 3  # Exceeding limit
+
+        # Create the column value with the sections
+        column_value = ColumnBlockFactory(sections=sections)
+
+        # Assemble the columns list with multiple column dictionaries
+        columns = [
+            {
+                "type": "column",
+                "value": column_value,
+            }
+        ] * 3
+
+        cls.main_menu.highlights = highlights
+        cls.main_menu.columns = columns
+
+    def test_main_menu_highlights_output_format(self):
+        """Test that main_menu_highlights outputs the correct format."""
+        highlights = main_menu_highlights({"request": self.mock_request}, self.main_menu)
+
+        self.assertIsInstance(highlights, list)
+        self.assertEqual(len(highlights), 3)
+
+        for highlight in highlights:
+            self.assertIn("text", highlight)
+            self.assertIn("description", highlight)
+            self.assertIn("url", highlight)
+
+    def test_main_menu_columns_output_format(self):
+        """Test that main_menu_columns outputs the correct format."""
+        columns = main_menu_columns({"request": self.mock_request}, self.main_menu)
+
+        self.assertIsInstance(columns, list)
+        self.assertEqual(len(columns), 3)
+
+        for column in columns:
+            self.assertIn("column", column)
+            self.assertIn("linksList", column)
+            self.assertIsInstance(column["linksList"], list)
+
+            for section in column["linksList"]:
+                self.assertIn("text", section)
+                self.assertIn("url", section)
+                self.assertIn("children", section)
+                self.assertIsInstance(section["children"], list)
+
+                for child in section["children"]:
+                    self.assertIn("text", child)
+                    self.assertIn("url", child)
```

Comment:
Can you do the full check? that is
`self.assertListEqual(columns, [...])` where `[...]` is the expected output.

Ditto for highlights

Reply:
Done! I think I implemented what you envisioned!

Reply:
@zerolab https://github.com/ONSdigital/dis-wagtail/pull/85/files#diff-bf6f55e1f68b6b5da79c34faa58386f8c997b00aa2637c7f2ac20207f9898a99R51